subsys/debug/tracing/cpu_stats.c

```diff
@@ -123,7 +121,6 @@ void sys_trace_thread_switched_out(void)
 	int key = irq_lock();
 
 	__ASSERT_NO_MSG(nested_interrupts == 0);
-	__ASSERT_NO_MSG(current_thread == k_current_get());
```

Comment:
If you need to allow NULL because of the first instance it would be better to do it the following way:
```
__ASSERT_NO_MSG(!current_thread || (current_thread == k_current_get()));
```
or even better i
```
__ASSERT_NO_MSG((!current_thread && first_out) || (current_thread == k_current_get()));
```
we also should add assert that checks switch out is not called twice before switch in.